pkg/compactor/block_upload.go

```diff
@@ -30,12 +34,21 @@ import (
 	util_log "github.com/grafana/mimir/pkg/util/log"
 )
 
-// Name of file where we store a block's meta file while it's being uploaded.
 const (
-	uploadingMetaFilename = "uploading-meta.json"
-	validationFilename    = "validation.json"
+	uploadingMetaFilename       = "uploading-meta.json" // Name of the file that stores a block's meta file while it's being uploaded
+	validationFilename          = "validation.json"     // Name of the file that stores a heartbeat time and possibly an error message
+	validationHeartbeatInterval = 1 * time.Minute       // Duration of time between heartbeats of an in-progress block upload validation
+	validationHeartbeatTimeout  = 5 * time.Minute       // Maximum duration of time to wait until a validation is able to be restarted
 )
 
+type BlockUploadConfig struct {
+	SkipValidation bool `yaml:"skip_validation" category:"experimental"`
```

Comment:
I wasn't sure at first if we want to make this visible to the user, but I think can be useful for some. :+1:

Reply:
Yeah I think this could be a nice option if the blocks are known to be valid (like in a migration) and you'd rather not process them.